public/friendlycode/js/fc/filesystem-sync.js

```diff
@@ -0,0 +1,104 @@
+define(["jquery"], function($) {
+  "use strict";
+
+  var FileSystemSync = {};
+
+  function pushFileChange(url, csrfToken, fs, path) {
+    var options = {
+      contentType: "application/json",
+      headers: {
+        "X-Csrf-Token": csrfToken
+      },
+      type: "PUT",
+      url: url
+    };
+
+    function send() {
+      var request = $.ajax(options);
+      request.done(function() {
+        if(request.readyState !== 4) {
+          return;
+        }
+
+        if(request.status !== 201 && request.status !== 200) {
+          // TODO: handle error case here
+          console.error("Server did not persist file");
```

Comment:
Prefix your console messages with `[Bramble]`.  Also, specify the `path` here, so we know which file it's talking about, and also the `request.status` you actually got:

```js
console.error("[Bramble] unable to persist `" + path + "` to server. Server responded with status `" + request.status + "`");
```